modules/machine-feature-aws-imds-options.adoc

```diff
@@ -0,0 +1,60 @@
+// Module included in the following assemblies:
+//
+// * machine_management/cluster_api_machine_management/cluster_api_provider_configurations/cluster-api-config-options-aws.adoc
+
+:_mod-docs-content-type: CONCEPT
+[id="machine-feature-aws-imds-options_{context}"]
+= Amazon EC2 Instance Metadata Service configuration options
+
+You can restrict the version of the Amazon EC2 Instance Metadata Service (IMDS) that machines on {aws-first} clusters use.
+Machines can require the use of link:https://docs.aws.amazon.com/AWSEC2/latest/UserGuide/configuring-instance-metadata-service.html[IMDSv2] (AWS documentation), or allow the use of IMDSv1 in addition to IMDSv2.
+
+////
+This is true but does not apply to TP clusters, reassess for Cluster API GA
+[NOTE]
+====
+To use IMDSv2 on AWS clusters that were created with {product-title} version 4.6 or earlier, you must update your boot image. 
+For more information, see "Updated boot images".
+====
+////
+
+include::snippets/apply-machine-configuration-method.adoc[tag=method-machine-template]
+
+[IMPORTANT]
+====
+Before creating machines that require IMDSv2, ensure that any workloads that interact with the IMDS support IMDSv2.
+====
+
+.Sample IMDS configuration
+[source,yaml]
+----
+apiVersion: infrastructure.cluster.x-k8s.io/v1beta2
+kind: AWSMachineTemplate
+# ...
+spec:
+  template:
+    spec:
+      instanceMetadataOptions:
+        httpEndpoint: enabled
+        httpPutResponseHopLimit: 1 <1>
+        httpTokens: optional <2>
+        instanceMetadataTags: disabled
+# ...
+----
+<1> Specifies the number of network hops allowed for IMDSv2 calls.
+If no value is specified, this parameter is set to `1` by default.
+<2> Specifies whether to require the use of IMDSv2.
+If no value is specified, this parameter is set to `optional` by default.
+The following values are valid:
+`optional`:: Allow the use of both IMDSv1 and IMDSv2.
+`required`:: Require IMDSv2.
+
+[NOTE]
+====
+The Machine API does not support the `httpEndpoint`, `httpPutResponseHopLimit`, and `instanceMetadataTags` fields. 
+If you migrate a Cluster API machine template that uses this feature to a Machine API compute machine set, any Machine API machines that it creates will not have these fields and the underlying instances will not use these settings. 
+Any existing machines that the migrated machine set manages will retain these fields and the underlying instances will continue to use these settings.
```

Comment:
@huali9 this is what @damdo and I landed on to explain this behavior. I feel comfortable merging this, but if you see any issues just let me know on Tuesday and I can make changes.